src/main/java/lavalink/client/io/jda/JdaLink.java

```diff
@@ -75,22 +70,17 @@ public JDA getJda() {
         return lavalink.getJdaFromSnowflake(String.valueOf(guild));
     }
 
-
-    private WebSocketClient getMainWs() {
-        return ((JDAImpl) getJda()).getClient();
-    }
-
     @Override
     protected void removeConnection() {
-        getMainWs().removeAudioConnection(guild);
```

Comment:
In JDA3 this would cancel a connection attempt (e.g. where it is pending due to rate-limiting). How does this work in JDA4?

Reply:
In JDA4 the connections are made via the DirectAudioController so JDA handles this for us.

Reply:
Why not cancel it from that then, rather than expecting the user to do it?

Reply:
I'm not sure what to do with this because we're not replacing the voice event handlers anymore

Reply:
I suppose I will have to investigate this when JDA4 comes out

The rest of the PR looks good to me :+1: